src/github/client/github-client.ts

```diff
@@ -70,6 +99,25 @@ export class GitHubClient {
 			... (gitHubConfig.proxyBaseUrl ? this.buildProxyConfig(gitHubConfig.proxyBaseUrl) : {})
 		});
 
+		// HOT-105065
+		if (certs) {
+			this.axios.interceptors.request.use(async (config: AxiosRequestConfig): Promise<AxiosRequestConfig> => {
+				try {
+					if (await booleanFlag(BooleanFlags.USE_CUSTOM_ROOT_CA_BUNDLE, gitHubConfig.baseUrl)) {
+						this.logger.info("Using proxy with custom CA");
+						config.httpsAgent = gitHubConfig.proxyBaseUrl
+							? new HttpsProxyAgentWithCustomCA(gitHubConfig.proxyBaseUrl, certs)
+							: new https.Agent({
```

Comment:
We don't need to support custom ca + non-proxy (our own server) situation?

Reply:
that's for local testing @gxueatlassian